### PR TITLE
Change `children` to `responses` in report api format

### PR DESCRIPTION
**Pull request by knowuh**
This PR is tiny ( cedf9de )  -- But its based on work in the report-link PR, so that work should be reviewed / integrated first.

The Client side changes are in this [portal-report PR](https://github.com/concord-consortium/portal-report/pull/1) which is just a little larger.

### CODE REVIEW

**Comment by pjanik** [comment]
I know this isn't too important, but what about using `answers` instead? The portal-report API uses `Answer` as a general keyword for this kind of object - we have lots of components with `Answer` prefix or suffix, e.g. `Answer`, `AnswerTable`, `ImageQuestionAnswer` etc. So, it would be a bit more consistent. I can make responses -> answers change in the client app, it would be one find-and-replace (simpler than previous children -> responses).

**Comment by knowuh** [comment]
@pjanik that sounds fine.  Shall I make the changes in the portal?

**Comment by knowuh** [comment]
I also added the Multiple Choice ID's to this PR,  seemed like to much overhead to add a PR just for that.

File: app/models/api/v1/report.rb [hunk 1]
```diff
@@ -93,6 +93,7 @@ def process_multiple_choice_answer(hash)
     # Make naming more consistent, otherwise we would have crazy sequence of [:answers][:answer][:answer] keys.
     hash[:answer] = hash[:answer].map do |a|
       {
+        id: a[:id],
```
Comment (scytacki) [inline app/models/api/v1/report.rb]: I'm not sure this `:id` is the right id here. It is hard to follow but I think the value of `a` for a multiple choice question comes from here: https://github.com/concord-consortium/rigse/blob/master/app/models/saveable/multiple_choice_answer.rb#L11
that value is cached in the report_learner#answers. So if we want to add the id then we need to update multiple_choice_answer and also add a "migration" to clear and rebuild the report_learner cache.  It might be better not do it as a real migration because it will be so dependent on all of the code. Instead it could be a rake task that we have to remember to run on the servers.

**Comment by knowuh** [comment]
@scytacki   2c0a009  fixes the broken spec.

**Comment by knowuh** [comment]
@scytacki commit 2c0a009  pulls the `response_id` from `Saveable::MultipleChoiceAnswer`, and makes it available to the `Report::Learner` and ultimately the `API::V1::Report`.  There already is a working take task to reprocess report_learners if we need it.  I took some notes in the commit.

**Comment by scytacki** [comment]
This looks good. The name of the class: `Saveable::MultipleChoiceRationaleChoice` threw me for a loop. After I dug a little I understand it now. 
Is it easy to update the branch so it only has commits that have not already been merged to master?

**Comment by knowuh** [comment]
There are some failing tests I am addressing right now.  I should have that wrapped up quickly, and then I will rebase.

**Comment by scytacki** [comment]
Looks like just one failing test left: https://travis-ci.org/concord-consortium/rigse/jobs/116405636#L1244-L1257

**Comment by scytacki** [comment]
:+1:
